FantasmoSDK/app/src/main/java/com/fantasmo/sdk/models/analytics/MotionManager.kt

```diff
@@ -39,30 +41,39 @@ class MotionManager(context: Context) : SensorEventListener {
     override fun onAccuracyChanged(sensor: Sensor?, accuracy: Int) {
     }
 
-    fun restart(){
-        if(!disabledSensor){
-            magneticField = MagneticField(0f,0f,0f)
-            magnetometer = sensorManager.getDefaultSensor(Sensor.TYPE_MAGNETIC_FIELD)
-            sensorManager.registerListener(this,magnetometer,SensorManager.SENSOR_DELAY_NORMAL)
-        }else{
-            magneticField = MagneticField(0f,0f,0f)
+    fun restart() {
+        magneticField = MagneticField(0f, 0f, 0f)
+        if (!disabledSensor) {
+            sensorManager = context.getSystemService(SENSOR_SERVICE) as SensorManager
+            // Verifying if Sensor Magnetic_Field exists
+            if (sensorManager.getDefaultSensor(Sensor.TYPE_MAGNETIC_FIELD) != null) {
+                //In positive case register it
+                magnetometer = sensorManager.getDefaultSensor(Sensor.TYPE_MAGNETIC_FIELD)
+                sensorManager.registerListener(
+                    this,
+                    magnetometer,
+                    SensorManager.SENSOR_DELAY_NORMAL
+                )
+                registered = true
+                Log.d(TAG, "Sensor Magnetic Field found")
+            } else {
+                //Else non existent
+                Log.d(TAG, "Sensor Magnetic Field not found")
+            }
         }
-
     }
 
-    fun stop(){
-        if(!disabledSensor){
-            magneticField = MagneticField(0f,0f,0f)
-            if(this::magnetometer.isInitialized){
-                sensorManager.unregisterListener(this,magnetometer)
+    fun stop() {
+        magneticField = MagneticField(0f, 0f, 0f)
+        if (!disabledSensor) {
+            if (registered) {
+                sensorManager.unregisterListener(this, magnetometer)
+                registered = false
             }
         }
-        else{
-            magneticField = MagneticField(0f,0f,0f)
-        }
     }
 
-    private fun disableSensor(){
+    private fun disableSensor() {
```

Comment:
From what I can see this function is never called - it's private and not called within this kotlin file. Also, the similarly named `disabledSensor` variable is always false (it's value never changes and it's also private).

Assuming private functions and vars cannot be called from outside the class, and only by the instance itself, is this behaving as intended?

Reply:
It may be usefull in the future if we have a use case where we need to disable the sensor.

Reply:
Also, forgot to mention, this method is invoked by unit tests, using reflection. In order to pass the tests we cannot emulate/simulate a sensor because it's hardware based and this allows us to test requests without the need of a sensor.